### DISCUSSION TITLE
Thonny using python3.12 on Raspberry pi

### DISCUSSION THREAD

**Discussion by ddodge40** [thonny/thonny#3093] [General]
When will Thonny for Raspberry Pi be upgraded to allow python3.12

**Comment by aivarannamaa** [comment]
Latest version of Thonny (4.1.4) already supports Python 3.12.

If you want to know, when Raspberry Pi OS puts Python 3.12 into their repository, then it's best to ask in their forum, but I suppose it won't happen before the next version of Debian comes out (probably in more than a year). Debian may choose Python 3.13 by then.

**Comment by ddodge40** [comment]
On a Raspberry Pi Thonny 4.1.4 only supports python 3.9.  It is buried in the notes.

I want to use the Match command that was added in Python 3.10.


Thanks


> On Jan 31, 2024, at 9:42 AM, Aivar Annamaa ***@***.*** ***@***.***>> wrote:
> 
> 
> Latest version of Thonny (4.1.4) already supports Python 3.12.
> 
> If you want to know, when Raspberry Pi OS puts Python 3.12 into their repository, then it's best to ask in their forum, but I suppose it won't happen before the next version of Debian comes out (probably in more than a year). Debian may choose Python 3.13 by then.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/thonny/thonny/discussions/3093#discussioncomment-8324480>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTPHL5ZOHWUVFSVTKYXWXLYRJ7B3AVCNFSM6AAAAABCOUUUTSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DGMRUGQ4DA>.
> You are receiving this because you authored the thread.
> 

Donald Dodge
***@***.*** ***@***.***>
W6SNX
Cell 714-271-7888

**Comment by ddodge40** [comment]
I have 3.12 on my pi and I can run within Python. But thonny will not workDon DodgeCell 714-271-7888W6SNXOn Jan 31, 2024, at 9:42 AM, Aivar Annamaa ***@***.***> wrote:﻿
Latest version of Thonny (4.1.4) already supports Python 3.12.
If you want to know, when Raspberry Pi OS puts Python 3.12 into their repository, then it's best to ask in their forum, but I suppose it won't happen before the next version of Debian comes out (probably in more than a year). Debian may choose Python 3.13 by then.

**Comment by aivarannamaa** [comment]
What does "will not work" mean in this case? Do you get an error message?

**Comment by ddodge40** [comment]
Thonny runs but only with Python 3.9 on my piDon DodgeCell 714-271-7888W6SNXOn Jan 31, 2024, at 12:38 PM, Aivar Annamaa ***@***.***> wrote:﻿
What does "will not work" mean in this case? Do you get an error message?

**Reply by ddodge40** [reply]
Thonny will only run with Python 3.9 on a raspberry pi
Thonny will run Python 3.12 on windows or Mac.  Question is when will Thonny run 3.12 on a pi

**Reply by aivarannamaa** [reply]
If your Raspberry Pi has Python 3.12, then you can make Thonny use it at "Tools => options => interpretrr"

**Comment by ddodge40** [comment]
Found the Python3.12 in opt/Python/bin and put that in options interpreter and it now works. I can now use the match case statements ThanksDon DodgeCell 714-271-7888W6SNXOn Feb 3, 2024, at 11:38 PM, Aivar Annamaa ***@***.***> wrote:﻿
If your Raspberry Pi has Python 3.12, then you can make Thonny use it at "Tools => options => interpretrr"
